feat: default post & attachment distribution settings

Takshil-Kunadia opened the pull request:
### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request, and the reason for such changes. -->

Closes https://app.asana.com/1/26890605006346/project/1210562189110460/task/1210629622176521?focus=true.

#### New Global Defaults for Distribution Settings:
This PR introduces two new global options in the Republication Tracker plugin settings to allow publishers to define default behaviors for content distribution:

- Default Status for Post Distribution
- Default Status for Attachment Distribution

#### Behavior:
- These options determine the default value (enabled or disabled) for:
    - The "Can distribute" field on attachments (affects newly uploaded attachments only).
    - The "Hide republication widget" field on posts.

- These settings do not affect existing posts or attachments, but only control the initial state for newly created items.

This gives publishers control over default behavior without manually toggling the option for every post or media file.

### How to test the changes in this Pull Request:

1. Navigate to the WordPress admin → Settings → Reading.
2. You’ll find two new settings at the bottom:
    - Default Post Distribution.
    - Default Attachment Distribution.
3. Enable/disable these options and save.
4. Upload a new media attachment.
    - Verify that the “Can distribute” checkbox reflects the selected default.
5. Create a new post.
    - Verify that the “Hide republication widget” option reflects the default.
6. Confirm that:
    - Changing the default affects only newly created content.
    - Existing posts and attachments retain their saved settings.

Takshil-Kunadia committed:
feat: add default post distribution setting

Takshil-Kunadia committed:
feat: add default attachment distribution setting

Takshil-Kunadia committed:
refactor: update settings labels

leogermani reviewed (changes requested):
Works great, can I just ask you to tweak the labels?

change "Enable default Post Distribution" to "Hide republication widgets on posts by default"

Move the other option just below the "Media Distribution" - if possible, make it part of the same section. and tweak the labels and description. Here's a mockup:

![image](https://github.com/user-attachments/assets/c70b4978-8ed8-445d-ad50-d81203f2f7e7)

Takshil-Kunadia committed:
style: move settings for clarity & update labels

Takshil-Kunadia commented:
Thanks! I’ve implemented the label changes as suggested and moved the media distributable setting just below the Media Distribution section. It now looks like this:

<img width="1352" alt="Screenshot 2025-06-27 at 11 50 28 AM" src="https://github.com/user-attachments/assets/e62592c5-2700-470c-aa7f-6cf486494d15" />

Does this placement feel okay to you, or does the extra space between the settings look off visually?

leogermani reviewed (approved):
perfect
